mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java

```diff
@@ -168,4 +168,20 @@ private DB extractDatabase(MongoClient mongo, MongoDBConfiguration config) {
 			}
 		}
 	}
+
+	protected MongoDBConfiguration getConfig() {
+		return config;
+	}
+
+	protected MongoClient getMongoClient() {
+		return mongo;
+	}
+
+	protected void setMongoClient(MongoClient mongoClient) {
```

Comment:
I prefer to make setter and getter protected rather than attributes to have control on how something is set or get from children.